test/token-service/test_tokenCreateTransaction.js

```diff
@@ -0,0 +1,8187 @@
+              },
+              {
+                type: "ecdsaSecp256k1PrivateKey",
+              },
+            ],
+          },
+          {
+            type: "keyList",
+            keys: [
+              {
+                type: "ecdsaSecp256k1PublicKey",
+              },
+              {
+                type: "ed25519PublicKey",
+              },
+            ],
+          },
+          {
+            type: "keyList",
+            keys: [
+              {
+                type: "ed25519PrivateKey",
+              },
+              {
+                type: "ecdsaSecp256k1PublicKey",
+              },
+            ],
+          },
+        ],
+      });
+      if (response.status === "NOT_IMPLEMENTED") {
+        this.skip();
+      }
+      const nestedKeyList = response.key;
+
+      response = await JSONRPCRequest("createToken", {
+        name: "testname",
+        symbol: "testsymbol",
+        treasuryAccountId: process.env.OPERATOR_ACCOUNT_ID,
+        feeScheduleKey: nestedKeyList,
+      });
+      if (response.status === "NOT_IMPLEMENTED") {
+        this.skip();
+      }
+
+      await verifyTokenCreationWithFeeScheduleKeyList(
+        response.tokenId,
+        nestedKeyList,
+      );
+    });
+
+    it("(#7) Creates a token with a valid ThresholdKey of ED25519 and ECDSAsecp256k1 private and public keys as its fee schedule key", async function () {
+      let response = await JSONRPCRequest("generateKey", {
+        type: "thresholdKey",
+        threshold: 2,
+        keys: [
+          {
+            type: "ed25519PrivateKey",
+          },
+          {
+            type: "ecdsaSecp256k1PublicKey",
+          },
+          {
+            type: "ed25519PublicKey",
+          },
+        ],
+      });
+      if (response.status === "NOT_IMPLEMENTED") {
+        this.skip();
+      }
+      const thresholdKey = response.key;
+
+      response = await JSONRPCRequest("createToken", {
+        name: "testname",
+        symbol: "testsymbol",
+        treasuryAccountId: process.env.OPERATOR_ACCOUNT_ID,
+        feeScheduleKey: thresholdKey,
+      });
+      if (response.status === "NOT_IMPLEMENTED") {
+        this.skip();
+      }
+
+      await verifyTokenCreationWithFeeScheduleKeyList(
+        response.tokenId,
+        thresholdKey,
+      );
+    });
+
+    it("(#8) Creates a token with an invalid key as its fee schedule key", async function () {
+      try {
+        const response = await JSONRPCRequest("createToken", {
+          name: "testname",
+          symbol: "testsymbol",
+          treasuryAccountId: process.env.OPERATOR_ACCOUNT_ID,
+          feeScheduleKey: crypto.randomBytes(88).toString("hex"),
+        });
+        if (response.status === "NOT_IMPLEMENTED") {
+          this.skip();
+        }
+      } catch (err) {
+        assert.equal(err.code, -32603, "Internal error");
+        return;
+      }
+
+      assert.fail("Should throw an error");
+    });
+  });
+
+  describe("Custom Fees", function () {
```

Comment:
We can: 
1. Extract the helper functions to different file in the **helpers** folder.
2. Merge common fee comparison logic with creating a generic helper functions to handle common comparison logic.
3. Write a single verifyTokenCreation function that handles the logic for comparing custom fees for different types, making it modular.
4. Call this general function for different fee types (Fixed, Fractional, Royalty) by providing the appropriate compare functions and types

The goal here is to enhance reusability and modularity, ensuring the code can be easily maintained and extended for future use.

Reply:
#260 can be used to investigate this.